Examples/build_all_examples.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+basename=$(dirname "$0")
+
+echo Cleaning libs...
+for lib in "$basename"/../lib*; do
+    make -C "$lib" clean
+done
+
+goodcount=0
+projectcount=0
+for project in "$basename"/*_test; do
+    echo -e "\e[34m========== BUILDING $project ==========\e[39m"
+    cd "$project"
+    make clean
+    if make all -j `nproc`; then
+        echo -e "\e[32m========== $project built successfully ==========\e[39m"
+        (( goodcount++ ))
+    else
+        echo -e "\e[31m========== Building $project failed! ==========\e[39m"
+    fi
+    (( projectcount++ ))
+    cd ..
+done
+
+echo -e "\e[34m
+BUILD COMPLETED
+[$goodcount/$projectcount of the projects built successfully]\e[39m"
```

Comment:
this would probably be nicer as a recursive make file - I think it can fairly easily be made to do this more tersely

Reply:
Good idea!

Reply:
I have just tried playing around with a makefile and it wasn't as easy as I thought... For some reason this:
```
%_test:
    $(MAKE) -C $@
```
...doesn't seem to find the folders and without pattern matching it ends up being quite clunky.

Also I deliberately made the script cleanbuild everything so that we can check if stuff has broken, so we don't want the makefile skipping any of the projects if they're already built.